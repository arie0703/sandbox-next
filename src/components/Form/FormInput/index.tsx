import React from 'react';
import { useFormContext } from 'react-hook-form';
import styles from '@/styles/components/form-input.module.scss';

export interface FormInputProps {
  formType: string
  memo?: React.ReactNode;
}

export const FormInput = ({ formType, memo }: FormInputProps) => {
  const { register } = useFormContext();

  switch (formType) {
    case 'nickname':
      return (
        <dl className={`${styles['_formInput']} ${styles['dl01']}`}>
          <dt className={styles['_title']}>ニックネーム</dt>
          <dd>
            <input
              id="nickname"
              type="text"
              {...register(formType)}
            />
            <span className={styles['memo']}>{memo}</span>
          </dd>
        </dl>
      );
    case 'residence':
      return (
        <dl className={`${styles['_formInput']} ${styles['dl01']}`}>
          <dt className={styles['_title']}>おところ</dt>
          <dd>
            <input
              id={formType}
              type="text"
              {...register(formType)}
            />
            <span className={styles['memo']}>{memo}</span>
          </dd>
        </dl>
      );
    case 'comment':
      return (
        <dl className={`${styles['_formInput']} ${styles['dl01']}`}>
          <dt className={styles['_title']}>コメント</dt>
          <dd>
            <input
              id={formType}
              type="text"
              {...register(formType)}
            />
            <span className={styles['memo']}>{memo}</span>
          </dd>
        </dl>
      );
    case 'icon-pattern':
      return (
        <dl className={`${styles['_formInput']} ${styles['dl01']}`}>
          <dt className={styles['_title']}>アイコン</dt>
          <dd>
            <select id={formType} {...register(formType)}>

              <option value="">アイコンを選択してください</option>
              <option value="A">A</option>
              <option value="B">B</option>
              <option value="C">C</option>
            </select>
          </dd>
        </dl>
      );
    default:
      return (
        <dl className={`${styles['_form_box']}`}>
          <dt className={styles['_title']}>{formType}</dt>
          <dd>
            <input
              id={formType}
              type="text"
            />
            <span className={styles['memo']}>{memo}</span>
          </dd>
        </dl>
      );
  }
};
