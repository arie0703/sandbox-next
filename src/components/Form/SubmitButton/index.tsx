import styles from '@/styles/components/submit-button.module.scss';

interface SubmitButtonProps {
  buttonText: string;
  handleClick: () => Promise<void>;
}

export const SubmitButton = ({ buttonText, handleClick }: SubmitButtonProps) => {

  return (
    <button className={styles['submitButton']} onClick={() => handleClick()}>
      {buttonText}
    </button>
  )
}
