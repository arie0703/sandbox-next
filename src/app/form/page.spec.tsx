import { render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import Form from "./page";

describe("Form", () => {
  it("フォームのレンダリング", async () => {
    render(<Form/>);
    expect(screen.getByText("Sample Form")).toBeInTheDocument();
  });

  it("スナップショットテスト", () => {
    const { asFragment } = render(<Form/>);
    expect(asFragment()).toMatchSnapshot();
  });
});
