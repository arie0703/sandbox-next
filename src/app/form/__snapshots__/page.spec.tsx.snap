// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form スナップショットテスト 1`] = `
<DocumentFragment>
  <div
    class="form"
  >
    <form>
      <h2>
        Sample Form
      </h2>
      <dl
        class="_formInput dl01"
      >
        <dt
          class="_title"
        >
          ユーザー名 
        </dt>
        <dd>
          <input
            id="user-name"
            name="user-name"
            placeholder="ユーザー名"
            type="text"
          />
          <span
            class="memo"
          />
        </dd>
      </dl>
    </form>
    <div
      class="submitButton"
    >
      Submit
    </div>
  </div>
</DocumentFragment>
`;
